src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
         var lo = input.ispace.bounds.lo:to_point()
         var hi = input.ispace.bounds.hi:to_point()
-        var n : int[dim]
+        var n : int[dim] --n is an array of size dim with the size of each dimension in the entries
         ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
-        var cufft_p : cufft_c.cufftHandle
-        var ok = cufft_c.cufftPlanMany(
-          &cufft_p,
-          dim,
-          &n[0],
-          [&int](0), 0, 0, -- inembed, istride, idist
-          [&int](0), 0, 0, -- onembed, ostride, odist
-          cufft_c.CUFFT_C2C,
-          1 -- batch
-        )
+      
+        --Call cufftPlanMany: cufftResult cufftPlanMany(cufftHandle *plan, int rank, int *n, int *inembed, int istride, int idist, int *onembed, int ostride, int odist, cufftType type, int batch) --rank = dimensionality of transform (1,2,3)
+        format.println("Calling cufftPlanMany...")
+
+        var ok = 0
+
+        if dtype_size == 8 and real_flag then
+          format.println("Calling cufftPlanMany with CUFFT_R2C: Float to Complex32 ...")
+          ok = cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_R2C, 1)
+        elseif dtype_size == 8 then
+          format.println("Calling cufftPlanMany with CUFFT_C2C: Complex32 to Complex32 ...")
+          ok = cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_C2C, 1)
+        elseif real_flag and dtype_size == 16 then
+          format.println("Calling cufftPlanMany with CUFFT_D2Z: Complex32 to Complex32 ...")
+          ok = cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_D2Z, 1)
+        elseif dtype_size == 16 then
+          format.println("Calling cufftPlanMany with CUFFT_Z2Z: Complex64 to Complex64 ... ...")
+          ok = cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_Z2Z, 1)
+        end
+
+        --Check return value of cufftPlanMany
+        if ok == cufft_c.CUFFT_INVALID_VALUE then
+          format.println("Invalid value in cufftPlanMany")
+        end
+
         regentlib.assert(ok == cufft_c.CUFFT_SUCCESS, "cufftPlanMany failed")
-        return cufft_p
-      else
+        format.println("cufftPlanMany Successful")
+      
+      --GPU not identified: return error
+      else 
+        format.println("GPU processor not identified: TOC_PROC not equal to processor kind")
         regentlib.assert(false, "make_plan_gpu must be executed on a GPU processor")
       end
     end
   end
 
-  -- Important: overwrites input/output!
-  __demand(__inline)
-  task iface.make_plan(input : region(ispace(itype), dtype),
-                       output : region(ispace(itype), dtype),
-                       plan : region(ispace(int1d), iface.plan))
+  --CPU version of make_plan. Calls make_plan_gpu if necessary
```

Comment:
Actually it's not the CPU version of `make_plan` because it is the user's entry point to both CPU and GPU functionality.

This is where it would be a good idea to write a proper documentation string.

Reply:
Wrote documentation string, edited description